extensions/aerospace/src/switchApps.tsx

```diff
@@ -0,0 +1,84 @@
+import { Action, ActionPanel, LaunchProps, List, Toast, getPreferenceValues, showToast } from "@raycast/api";
+import { Windows, focusWindow, getWindows } from "./utils/appSwitcher";
+import { useEffect, useMemo, useState } from "react";
+import { useCachedState } from "@raycast/utils";
+
+interface Preferences {
+  config: string;
+  defaultWorkspace: string;
+}
+
+export default function Command(props: LaunchProps<{ arguments: Arguments.SwitchApps }>) {
+  const { defaultWorkspace } = getPreferenceValues<Preferences>();
+
+  const [windows, setWindows] = useCachedState<Windows>("windows", []);
+
+  const workspace = props.arguments.workspace ? props.arguments.workspace : defaultWorkspace;
+
+  useEffect(() => {
+    const updatedWindows = getWindows(workspace);
+    setWindows(updatedWindows);
+    console.log({ updatedWindows });
+  }, []);
+
+  const [error, setError] = useState<Error>();
+
+  useEffect(() => {
+    setTimeout(() => {
+      setError(new Error("Booom 💥"));
+    }, 20000);
```

Comment:
**question:** Is this a leftover code that we'll delete? I couldn't get the benefit of these lines.

Reply:
I copied this from another extension. To set an error if the command gets stuck.

PS: Not a developer so if there is a better way or if it's better to remove it's ok with me.

Reply:
https://developers.raycast.com/information/best-practices#handle-errors

Reply:
I wasn't aware of this, thanks for the docs link!